services/director-v2/src/simcore_service_director_v2/utils/dask_client_utils.py

```diff
@@ -193,38 +198,47 @@ async def get_gateway_auth_from_params(
 _PING_TIMEOUT_S: Final[int] = 5
 
 
-async def test_gateway_endpoint(
+async def test_scheduler_endpoint(
     endpoint: AnyUrl, authentication: ClusterAuthentication
 ) -> None:
     """This method will try to connect to a gateway endpoint and raise a ConfigurationError in case of problem
 
     :raises ConfigurationError: contians some information as to why the connection failed
     """
     try:
-        gateway_auth = await get_gateway_auth_from_params(authentication)
-        async with dask_gateway.Gateway(
-            address=f"{endpoint}", auth=gateway_auth, asynchronous=True
-        ) as gateway:
-            # this does not yet create any connection to the underlying gateway.
-            # since using a fct from dask gateway is going to timeout after a long time
-            # we bypass the pinging by calling in ourselves with a short timeout
-            async with httpx.AsyncClient(
-                transport=httpx.AsyncHTTPTransport(retries=2)
-            ) as client:
-                # try to get something the api shall return fast
-                response = await client.get(
-                    f"{endpoint}/api/version", timeout=_PING_TIMEOUT_S
-                )
-                response.raise_for_status()
-                # now we try to list the clusters to check the gateway responds in a sensible way
-                await gateway.list_clusters()
-
-            logger.debug("Pinging %s, succeeded", f"{endpoint=}")
+        if _is_internal_scheduler(authentication):
+            client = await distributed.Client(
```

Comment:
WARNING: 
- is this ``client`` (``dask.distributed.Client``) not being shadowed by the ``client`` below? (``httpx.AsyncClient``)??
- you do not close the ``dask.distributed.Client``?

Reply:
so there is no shadowing, but you are right I shall close it correctly. Thanks!